docs/data-hub/dataframes.md

```diff
@@ -0,0 +1,130 @@
+# Using Deep Origin DataFrames
+
+This page describes how to use Deep Origin DataFrames, which are the primary object you will use to interact with databases on Deep Origin. This page will cover:
+
+- fetching data from a Deep Origin database
+- modifying data locally
+- writing data back to Deep Origin
+
+
+!!! question "What is a Deep Origin DataFrame?"
+    A Deep Origin DataFrame is a subclass of a [pandas.DataFrame](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html) that is created from a Deep Origin database, and can easily write data back to it. Because it is a subclass of a pandas DataFrame, all pandas DataFrame methods work on Deep Origin DataFrames. 
+
+## Create a DataFrame
+
+Create a DataFrame using:
+
+```python
+from deeporigin import DataFrame
+df = DataFrame.from_deeporigin("database-id")
+```
+
+In a Jupyter Lab, you should be able to view the DataFrame using:
+
+```py
+df
+```
+
+which should show you something like this:
+
+![DataFrame](../images/dataframe-0.png)
+
+
+!!! tip "Information in the DataFrame"
+    In addition to information you would find in the rows and columns of a pandas DataFrame, a Deep Origin DataFrame also contains metadata about the underlying database. In the view above we also see:
+
+    - The name of the database on Deep Origin.
+    - A link to the database on Deep Origin.
+    - When the database was created.
+    - Information about the last edit made to the database. 
+
+
+## Modify data in the DataFrame
+
+Because a Deep Origin DataFrame is a subclass of a pandas DataFrame, all pandas DataFrame methods work on Deep Origin DataFrames. In this example, we modify values in one of the columns, or modify a single cell.
+
+
+=== "Modify entire columns"
+
+    To modify entire columns, use native pandas syntax:
+
+    ```python
+    df["y^2"] = df["y"] ** 2
+    ```
+
+=== "Modify data in a single cell"
+
+    To modify data in a single cell, use native pandas syntax (the at operator):
+
+    ```python
+    df.at["sgs-1", "y"] = 10
+    ```
+
+In either case, when we view the DataFrame once more using `df`, we see the updated data, together with a warning telling us that we have local changes that haven't been written back to Deep Origin.
+
+
+![DataFrame with warning telling us about local changes that have not been written back to Deep Origin](../images/dataframe-1.png)
+
+## Write data back to Deep Origin
+
+### Using the `to_deeporigin` method
```

Comment:
Either at the beginning or end of this section, I think it would be helpful to outline what `to_deeporigin` can and cannot do:
- Update values of existing cells
  - Non-files
  - Files
- Update names of existing columns
- Update types of existing columns
- Create new rows
- Create new columns
- Remove rows
- Remove columns
- Create database

This could simply be a callout with a link to the "limitations" section below

My aim is simply to provide the user clarify so they know what to expect; we don't need to support all of the above.

Reply:
fixed